copyfromstaging.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -ex
+
+TMP=$(mktemp -d)
+
+copy_one_api() {
+  PACKAGE=$1
+  STAGING_DIR=owl-bot-staging/$1
+  PACKAGE_DIR=apis/$1
+
+  # All source files need the extension .g.cs.  Some will only be generated with the
+  # extension .cs.  Rename files that end only in .cs.
+  for cs_file in $(find "$STAGING_DIR" -name '*.cs'); do
+    if [[ $cs_file != *.g.cs ]]; then
+      mv $cs_file $(echo $cs_file | sed -E 's/.cs$/.g.cs/g')
+    fi
+  done
+
+  # Remove the newly generated standalone snippets until they are ready for surfacing.
+  rm -rf "$STAGING_DIR/$1.StandaloneSnippets"
+
+  # We generate our own project files
+  rm -f $(find "$STAGING_DIR" -name '*.csproj')
+
+  # Files generated by microgenerator are mysteriously marked as executable.
+  # They shouldn't be.
+  chmod a-x $(find "$STAGING_DIR" -name '*.cs')
+
+  # Wipe out all the existing source files we're about to replace.
```

Comment:
Is there a reason to do this "manually" rather than within .OwlBot.yaml? (I don't mind which way it happens - I'm just trying to understand the reasoning. Is it an ordering issue?)

Reply:
If we use a staging directory, then I like to break the logic into two steps with a clear division of responsibilities:
1.  Copy the full generated source tree.  (owl-bot copy)
2.  Cherry-pick and modify the code.  (owl-bot post processor)